core/trino-main/src/main/java/io/trino/metadata/MetadataUtil.java

```diff
@@ -225,6 +226,11 @@ public static void checkRoleExists(Session session, Node node, Metadata metadata
             }
         }
         catalog.ifPresent(catalogName -> getRequiredCatalogHandle(metadata, session, node, catalogName));
+
+        if (catalog.isPresent() && !metadata.isCatalogManagedSecurity(session, catalog.get())) {
+            throw semanticException(NOT_SUPPORTED, node, "Catalog '%s' does not support role management", catalog.get());
```

Comment:
Can you please provide tests that now statements like `SET ROLE x IN CATALOG c` fails with this error when connector access control does not exists for given catalog?